plugins/nodes/src/js/data/__tests__/NetworkNodesClient-test.ts

```diff
@@ -0,0 +1,73 @@
+const mockRequest = jest.fn();
+jest.mock("@dcos/http-service", () => ({
+  request: mockRequest
+}));
+
+import { from } from "rxjs";
+import { marbles } from "rxjs-marbles/jest";
+import { take } from "rxjs/operators";
+
+import * as NetworkNodesClient from "../NetworkNodesClient";
+
+describe("Network Nodes Client", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("makes a request", () => {
+    mockRequest.mockReturnValueOnce(from([{}]));
+    NetworkNodesClient.fetchNetworkNodes();
+    expect(mockRequest).toHaveBeenCalled();
+  });
+
+  it(
+    "emits an event when the data is received",
+    marbles(m => {
+      const expectedResult = [
+        JSON.stringify({
+          updated: "2019-02-04T14:17:12.697Z",
+          public_ips: ["18.185.33.115"],
+          private_ip: "10.0.6.192",
+          hostname: "ip-10-0-6-192"
+        })
+      ];
+      const expected$ = m.cold("--(j|)", {
+        j: {
+          response: expectedResult,
```

Comment:
are we expecting array of strings? `const expectedResult = [JSON.stringify(...)]` or do I read this incorrectly?

Reply:
addressed